cranelift-faerie/src/backend.rs

```diff
@@ -87,6 +247,174 @@ impl FaerieCompiledFunction {
     }
 }
 
+struct CFIEncoder<'a> {
+    reg_map: &'a DwarfRegMapper<'a>,
+    cfa_def_reg: Option<isa::RegUnit>,
+    cfa_def_offset: Option<isize>,
+}
+
+struct DwarfRegMapper<'a> {
+    isa: &'a Box<dyn TargetIsa>,
+    reg_info: RegInfo,
+}
+
+impl<'a> DwarfRegMapper<'a> {
+    pub fn for_isa(isa: &'a Box<dyn TargetIsa>) -> Self {
+        DwarfRegMapper {
+            isa,
+            reg_info: isa.register_info(),
+        }
+    }
+
+    /// Translate a Cranelift `RegUnit` to its matching `Register` for DWARF use.
+    ///
+    /// panics if `reg` cannot be translated - the requested debug information would be
+    /// unencodable.
+    pub fn translate_reg(&self, reg: RegUnit) -> gimli::Register {
+        match self.isa.name() {
+            "x86" => {
+                const X86_GP_REG_MAP: [u16; 16] = [
+                    // cranelift rax == 0 -> dwarf rax == 0
+                    0, // cranelift rcx == 1 -> dwarf rcx == 2
```

Comment:
Please use [`gimli::X86_64`](https://docs.rs/gimli/0.19.0/gimli/struct.X86_64.html)

Reply:
didn't know that existed! excellent.